### PR TITLE
run_qemu.sh: clarify exit status in EXIT trap handler

### PR DESCRIPTION
**Pull request by marc-hb**
This is all incredibly subtle, try to dumb it down.

Also: guard against exit code pollution from cleanup()

### CODE REVIEW

**Review by stellarhopper — APPROVED** [review]

**Comment by marc-hb** [comment]
mkosi error code clearly printed with `run_qemu.sh --debug` in https://github.com/pmem/run_qemu/actions/runs/12172092323/job/33950154580?pr=71

**Review by stellarhopper — APPROVED** [review]
